### PR TITLE
Add scroll-shadow modifier

### PR DESCRIPTION
**Pull request by JulienVannier66**
### What does this PR do?

Add scroll-shadow modifier. It detects if a container is scrollable. If truthy, add `scroll-shadow` class

Related to: nothing

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [ ] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/modifiers/scroll-shadow.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      state.resizeObserver = new ResizeObserver((entries: ResizeObserverEntry[]) => {
+        const observedEntry = entries.find((entry) => {
+          return entry.target === element;
+        });
+
+        if (observedEntry) {
+          state.hasScrollbar = observedEntry.target.scrollHeight > observedEntry.target.clientHeight;
+
+          if (state.hasScrollbar && !state.element.classList.contains('scroll-shadow')) {
+            state.element.classList.add('scroll-shadow', `scroll-shadow--${color || 'white'}`);
```
Comment (Miexil) [inline addon/modifiers/scroll-shadow.ts:32]: To future proof a little, you may want tot define either a color getter, or a CSSclass getter !

File: addon/components/o-s-s/infinite-select.hbs [hunk 2]
```diff
@@ -6,51 +6,52 @@
...
             {{else}}
-              {{t "oss-components.infinite-select.empty"}}
+              {{get item this.itemLabel}}
             {{/if}}
+          </li>
+        {{else}}
+          <div class="fx-col fx-xalign-center">
+            <img
+              src="/@upfluence/oss-components/assets/images/empty-state-skeleton.png"
+              alt="Empty content" width="15%" />
```
Comment (phndiaye) [inline addon/components/o-s-s/infinite-select.hbs:32]: trad here :p

File: addon/modifiers/scroll-shadow.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    colorCSSClass(args: any) {
+      const color: ColorAvailable = args.named.color;
+      return `scroll-shadow--${color || 'default'}`;
+    },
+
+    installModifier(state: State, element: Element, args: any) {
+      state.element = element;
+
+      state.resizeObserver = new ResizeObserver((entries: ResizeObserverEntry[]) => {
+        window.requestAnimationFrame(() => {
```
Comment (JulienVannier66) [inline addon/modifiers/scroll-shadow.ts:34]: Without this, tests fail with `ResizeObserver - loop limit exceeded`. I found this fix in https://stackoverflow.com/questions/49384120/resizeobserver-loop-limit-exceeded

File: addon/modifiers/scroll-shadow.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  () => ({
+    createModifier() {
+      return {
+        element: null
+      };
+    },
+
+    baseCSSClass() {
+      return 'scroll-shadow';
+    },
```
Comment (phndiaye) [inline addon/modifiers/scroll-shadow.ts:23]: why not use a `const` outside of the modifier manager for this ?

File: addon/modifiers/scroll-shadow.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    colorCSSClass(args: any) {
```
Comment (phndiaye) [inline addon/modifiers/scroll-shadow.ts:25]: maybe just pass it the `args.named.color` from outside, and move this to: `colorCSSClass(color: string)` ?

**Review by phndiaye — APPROVED** [review]

**Review by Miexil — APPROVED** [review]

**Pull request merged**